lib/email_alert_signup_api.rb

```diff
@@ -62,12 +62,26 @@ def massaged_attributes
     @massaged_attributes ||= attributes.dup.tap do |massaged_attributes|
       available_choices.each do |choice|
         key = choice["filter_key"] || choice["facet_id"]
-        value = (massaged_attributes['filter'] || {})[key]
-        next unless value
+        value = massaged_attributes.dig('filter', key)
+        next if value.nil?
 
-        massaged_attributes[key] = value
+        massaged_attributes[key] = parsed_filter_value(choice, value)
       end
       massaged_attributes.delete("filter")
     end
   end
+
+  def parsed_filter_value(choice, value)
+    chosen_options = Array(value)
+    # filter_values can be set if we want to send an array of values to
+    # the email-alert-api. E.g. document group: research, translates to
+    # to several document types being sent to rummager / email-alert-api.
+    chosen_facet_choices = choice.fetch('facet_choices', []).select { |option|
+      chosen_options.include?(option['key'])
+    }
+
+    chosen_facet_choice_values = chosen_facet_choices.map { |option| option['filter_values'] }.flatten
+
```

Comment:
This took me a while to figure out, I'd suggest (but really, only suggest rather than demand) that this might be slightly easier to comprehend
```
    chosen_facet_choice_values = choice.fetch('facet_choices', []).map { |option|
      if chosen_options.include?(option['key'])
        option['filter_values']
      end
    }

    chosen_facet_choice_values.flatten!
    chosen_facet_choice_values.compact!
```